### ISSUE TITLE
HttpPathModifier can't be serialized / deserialized

### ISSUE THREAD

**Issue by rthomas** [linkerd/k8s-gateway-api-rs#39]
When attempting to serialize an `HttpRoute` containing an `HttpPathModifier` a runtime error is thrown by serde due to the tagging being used on a newtype variant that does not contain a struct or map.

```
Error("cannot serialize tagged newtype variant HttpPathModifier::ReplacePrefixMatch containing a string", line: 0, column: 0)'
```

From the serde docs (https://serde.rs/enum-representations.html):

> This representation works for struct variants, newtype variants containing structs or maps, and unit variants but does not work for enums containing tuple variants. 

This can be reproduced by trying to serialize this struct:

```rust
        let r = HttpRoute {
            metadata: ObjectMeta {
                name: Some("route_name".to_string()),
                ..ObjectMeta::default()
            },
            spec: HttpRouteSpec {
                rules: Some(vec![HttpRouteRule {
                    filters: Some(vec![HttpRouteFilter::URLRewrite {
                        url_rewrite: HttpUrlRewriteFilter {
                            hostname: None,
                            path: Some(HttpPathModifier::ReplacePrefixMatch("/".to_string())),
                        },
                    }]),
                    matches: None,
                    backend_refs: None,
                }]),
                inner: CommonRouteSpec { parent_refs: None },
                hostnames: None,
            },
            status: None,
        };
```

**Comment by rthomas** [comment]
This was changed in this PR https://github.com/linkerd/k8s-gateway-api-rs/pull/8

**Comment by kleimkuhler** [comment]
Closed by #40
